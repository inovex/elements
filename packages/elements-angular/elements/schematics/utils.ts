import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
