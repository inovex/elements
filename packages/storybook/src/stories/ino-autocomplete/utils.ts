import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
