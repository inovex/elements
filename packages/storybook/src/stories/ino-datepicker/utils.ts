import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
