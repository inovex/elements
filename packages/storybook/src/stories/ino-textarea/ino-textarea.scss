.story-textarea {
  display: flex;
  justify-content: space-around;
  ino-textarea {
    margin-top: 15px;
    margin-bottom: 15px;
  }
}

