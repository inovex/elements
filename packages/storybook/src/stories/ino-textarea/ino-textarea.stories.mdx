import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-textarea/readme.md";

<Meta title="Input/ino-textarea/Documentation" />

<Description markdown={readme} />
