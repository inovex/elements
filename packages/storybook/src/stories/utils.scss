/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
