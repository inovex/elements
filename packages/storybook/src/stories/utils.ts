import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
