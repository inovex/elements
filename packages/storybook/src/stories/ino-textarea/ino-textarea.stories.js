import { boolean, number, text } from '@storybook/addon-knobs';

import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import componentReadme from '_local-elements/src/components/ino-textarea/readme.md';
import './ino-textarea.scss';

// https://github.com/storybooks/storybook/issues/4337#issuecomment-428495664
function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function (e) {
    const el = e.target;
    if (el.tagName.toLowerCase() !== 'ino-textarea') {
      return;
    }

    el.setAttribute('value', e.detail);
  };

  document.addEventListener('valueChange', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('valueChange', eventHandler);
  };
}

export default {
  title: 'Input/<ino-textarea>',
  parameters: {
    actions: {
      handles: ['input .customizable-textarea', 'valueChange .customizable-textarea']
    }
  },
  decorators: [
    withStencilReadme(componentReadme),
    (story) => {
      subscribeToComponentEvents();
      return story();
    },
  ],
};

export const DefaultUsage = () => /*html*/ `
<div class="story-textarea">
    <style>
      ino-textarea.customizable-textarea {
        --textarea-outline-color: ${text(
          '--textarea-outline-color',
          '#3d40f5',
          'Custom Properties'
        )};
        --textarea-caret-color: ${text('--textarea-caret-color', '#3d40f5', 'Custom Properties')};
        --textarea-label-color: ${text('--textarea-label-color', '#3d40f5', 'Custom Properties')};
      }
  </style>
  <ino-textarea class="customizable-textarea"
    cols="${number('cols', 60)}"
    rows="${number('rows', 5)}"
    placeholder="${text('placeholder', '')}"
    value="${text('value', '')}"
    ino-label="${text('ino-label', 'Customizable textarea')}"
    minlength="${number('minlength', 0)}"
    maxlength="${number('maxlength', 30)}"
    disabled="${boolean('disabled', false)}"
    required="${boolean('required', false)}"
    ino-show-label-hint="${boolean('ino-show-label-hint', false)}"
    autogrow="${boolean('autogrow', false)}">
  </ino-textarea>

  <h4>Labels</h4>
  <ino-textarea ino-label="Floating label" cols="30" rows="3"></ino-textarea>
  <ino-textarea ino-label="Floating label" value="With value" cols="30" rows="3"></ino-textarea>

  <h4>States</h4>
  <ino-textarea placeholder="Disabled" disabled cols="30" rows="3"></ino-textarea>
  <ino-textarea ino-label="Required" required ino-show-label-hint cols="30" rows="3"></ino-textarea>
  <ino-textarea ino-label="Optional" cols="30" ino-show-label-hint rows="3"></ino-textarea>
</div>
`;

DefaultUsage.storyName = 'Default usage';
