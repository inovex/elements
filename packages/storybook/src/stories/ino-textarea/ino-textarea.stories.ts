import { html } from 'lit-html';
import { useEffect } from '@storybook/client-api';
import { defaultDecorator } from '../utils';
import { Story } from '@storybook/web-components';
import { Components } from '@inovex.de/elements';
import './ino-textarea.scss';

export default {
  title: 'Input/ino-textarea',
  component: 'ino-textarea',
  parameters: {
    actions: {
      handles: [
        'input .customizable-textarea',
        'valueChange .customizable-textarea',
      ],
    },
  },
  decorators: [
    (story) => defaultDecorator(story, 'story-textarea'),
    (story) => {
      useEffect(() => {
        const eventHandler = (e) => e.target.setAttribute('value', e.detail);
        const textareas = document.querySelectorAll('ino-textarea');
        textareas.forEach((t) =>
          t.addEventListener('valueChange', eventHandler)
        );
        return () =>
          textareas.forEach((t) =>
            t.removeEventListener('valueChange', eventHandler)
          );
      });
      return story();
    },
  ],
};

export const Playground: Story<Components.InoTextarea> = (args) => html`
  <ino-textarea
    class="customizable-textarea"
    cols="${args.cols}"
    rows="${args.rows}"
    placeholder="${args.placeholder}"
    value="${args.value}"
    label="${args.label}"
    outline="${args.outline}"
    minlength="${args.minlength}"
    maxlength="${args.maxlength}"
    disabled="${args.disabled}"
    required="${args.required}"
    show-label-hint="${args.showLabelHint}"
    autogrow="${args.autogrow}"
    show-character-counter="${args.showCharacterCounter}"
  ></ino-textarea>
`;
Playground.args = {
  cols: 60,
  rows: 5,
  placeholder: '',
  value: '',
  label: 'Label',
  minlength: 0,
  maxlength: 30,
  disabled: false,
  required: false,
  outline: true,
  showLabelHint: false,
  autogrow: false,
  showCharacterCounter: true,
};

export const Labels = () => html`
  <ino-textarea label="Floating label" cols="30" rows="3"></ino-textarea>
  <ino-textarea
    label="Floating label"
    value="With value"
    cols="30"
    rows="3"
  ></ino-textarea>
`;

export const Filled = () => html`
  <ino-textarea
    label="Filled"
    cols="50"
    rows="3"
    outline="false"
  ></ino-textarea>
`;

export const Outline = () => html`
  <ino-textarea label="Outline" cols="50" rows="3" outlined></ino-textarea>
`;

export const States = () => html`
  <ino-textarea
    placeholder="Disabled"
    disabled
    cols="30"
    rows="3"
  ></ino-textarea>
  <ino-textarea
    label="Required"
    required
    show-label-hint
    cols="30"
    rows="3"
  ></ino-textarea>
  <ino-textarea
    label="Optional"
    cols="30"
    show-label-hint
    rows="3"
  ></ino-textarea>
`;

export const Autogrow = () => html`
  <ino-textarea
    label="Autogrowing textarea"
    autogrow
    class="autogrow-textarea"
    rows="1"
    cols="50"
    outlined="false"
  ></ino-textarea>
`;

export const CharacterCount = () => html`
  <ino-textarea
    label="Character count"
    autogrow
    rows="1"
    cols="50"
    outlined="false"
    maxlength="200"
    show-character-counter="true"
  ></ino-textarea>
`;

//     <style>
//       ino-textarea.customizable-textarea {
//         --ino-textarea-outline-color: ${text(
//   '--ino-textarea-outline-color',
//   '#820F35',
//   'Custom Properties'
// )};
//         --ino-textarea-caret-color: ${text('--ino-textarea-caret-color', '#820F35', 'Custom Properties')};
//         --ino-textarea-label-color: ${text('--ino-textarea-label-color', '#820F35', 'Custom Properties')};
//         --ino-textarea-floated-label-color: ${text('--ino-textarea-floated-label-color', '#820F35', 'Custom Properties')};
//         --ino-textarea-line-color: ${text('--ino-textarea-line-color', '#820F35', 'Custom Properties')};
//       }
//   </style>
