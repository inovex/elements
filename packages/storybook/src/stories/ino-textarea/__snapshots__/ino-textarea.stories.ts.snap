// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-textarea Autogrow smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea class="customizable-textarea ino-textarea--outline hydrated"
                  cols="60"
                  rows="5"
                  label="Label"
                  outline="true"
                  minlength="0"
                  disabled="false"
                  required="false"
                  show-label-hint="false"
                  autogrow="true"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined">
        <textarea class="mdc-text-field__input"
                  cols="60"
                  minlength="0"
                  rows="5"
                  style="overflow: hidden; overflow-wrap: break-word; height: 120px;"
        >
        </textarea>
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style
              >
                Label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;

exports[`Input/ino-textarea CharacterCount smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea class="customizable-textarea ino-textarea--outline hydrated"
                  cols="60"
                  rows="5"
                  label="Label"
                  outline="true"
                  minlength="0"
                  maxlength="30"
                  disabled="false"
                  required="false"
                  show-label-hint="false"
                  autogrow="false"
                  show-character-counter="true"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined mdc-text-field--with-internal-counter">
        <textarea class="mdc-text-field__input"
                  cols="60"
                  maxlength="30"
                  minlength="0"
                  rows="5"
        >
        </textarea>
        <div class="mdc-text-field-character-counter">
          0 / 30
        </div>
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style="transition-duration: 0s;"
              >
                Label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;

exports[`Input/ino-textarea Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea class="customizable-textarea ino-textarea--outline hydrated"
                  cols="60"
                  rows="5"
                  label="Label"
                  outline="true"
                  minlength="0"
                  disabled="true"
                  required="false"
                  show-label-hint="false"
                  autogrow="false"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined mdc-text-field--disabled">
        <textarea class="mdc-text-field__input"
                  cols="60"
                  disabled
                  minlength="0"
                  rows="5"
        >
        </textarea>
        <ino-label class="ino-label--outlined ino-label--disabled hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style
              >
                Label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;

exports[`Input/ino-textarea Labels smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea label="Floating label"
                  cols="30"
                  rows="3"
                  class="ino-textarea--outline hydrated"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined">
        <textarea class="mdc-text-field__input"
                  cols="30"
                  rows="3"
        >
        </textarea>
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style
              >
                Floating label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
    <ino-textarea label="Floating label"
                  value="With value"
                  cols="30"
                  rows="3"
                  class="ino-textarea--outline hydrated"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined mdc-text-field--label-floating">
        <textarea class="mdc-text-field__input"
                  cols="30"
                  rows="3"
        >
        </textarea>
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded mdc-notched-outline--notched">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch"
                 style="width: 80px;"
            >
              <label class="mdc-floating-label mdc-floating-label--float-above"
                     style
              >
                Floating label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;

exports[`Input/ino-textarea Outline smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea class="customizable-textarea ino-textarea--outline hydrated"
                  cols="60"
                  rows="5"
                  label="Label"
                  outline="true"
                  minlength="0"
                  disabled="false"
                  required="false"
                  show-label-hint="false"
                  autogrow="false"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined">
        <textarea class="mdc-text-field__input"
                  cols="60"
                  minlength="0"
                  rows="5"
        >
        </textarea>
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style
              >
                Label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;

exports[`Input/ino-textarea Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea class="customizable-textarea ino-textarea--outline hydrated"
                  cols="60"
                  rows="5"
                  label="Label"
                  outline="true"
                  minlength="0"
                  maxlength="30"
                  disabled="false"
                  required="false"
                  show-label-hint="false"
                  autogrow="false"
                  show-character-counter="true"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined mdc-text-field--with-internal-counter">
        <textarea class="mdc-text-field__input"
                  cols="60"
                  maxlength="30"
                  minlength="0"
                  rows="5"
        >
        </textarea>
        <div class="mdc-text-field-character-counter">
          0 / 30
        </div>
        <ino-label class="ino-label--outlined hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style
              >
                Label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;

exports[`Input/ino-textarea Required smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea class="customizable-textarea ino-textarea--outline hydrated"
                  cols="60"
                  rows="5"
                  label="Label*"
                  outline="true"
                  minlength="0"
                  disabled="false"
                  required="true"
                  show-label-hint="false"
                  autogrow="false"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined">
        <textarea class="mdc-text-field__input"
                  cols="60"
                  minlength="0"
                  required
                  rows="5"
        >
        </textarea>
        <ino-label class="ino-label--outlined ino-label--required hydrated">
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label mdc-floating-label--required"
                     style
              >
                Label*
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;

exports[`Input/ino-textarea ShowLabelHint smoke-test 1`] = `
<div id="root-inner">
  <div class="story-textarea">
    <ino-textarea class="customizable-textarea ino-textarea--outline hydrated"
                  cols="60"
                  rows="5"
                  label="Label"
                  outline="true"
                  minlength="0"
                  disabled="false"
                  required="false"
                  show-label-hint="true"
                  autogrow="false"
    >
      <div class="mdc-text-field mdc-text-field--textarea mdc-text-field--outlined">
        <textarea class="mdc-text-field__input"
                  cols="60"
                  minlength="0"
                  rows="5"
        >
        </textarea>
        <ino-label show-hint
                   class="ino-label--show-hint ino-label--outlined hydrated"
        >
          <div class="mdc-notched-outline mdc-notched-outline--upgraded">
            <div class="mdc-notched-outline__leading">
            </div>
            <div class="mdc-notched-outline__notch">
              <label class="mdc-floating-label"
                     style="transition-duration: 0s;"
              >
                Label
              </label>
            </div>
            <div class="mdc-notched-outline__trailing">
            </div>
          </div>
        </ino-label>
      </div>
    </ino-textarea>
  </div>
</div>
`;
